config_templates/pool.yaml

```diff
@@ -129,3 +129,6 @@ pool_specification:
       - kata_containers
       - singularity
     default: null
+  environment_variables:
```

Comment:
Thinking about this a little more, I think the proper place for these two properties is too nest them under a new property while moving the other node prep property around. As these environment variables only apply to the "start task" (or "node prep" in Batch Shipyard), it doesn't quite make sense to put it at the general pool level as some users may get confused that these environment variables apply globally to pool nodes (e.g., logging in with an SSH user and these env vars aren't there).

Right now we have `additional_node_prep_commands`. I think it will be best to re-organize as:

```yaml
pool_specification:
  # ... stuff
  additional_node_prep:
    commands:
       pre: []
       post: []
    environment_variables:
      abc: 'xyz'
    environment_variables_keyvault_secret_id: https://myvault.vault.azure.net/secrets/nodeprep-envvar
```

This is trending to be a little out of scope for this PR, so I can take this on separately if you want; you'd just need to be aware of a breaking change later.

Reply:
That sounds sensible - it would be great if you could do that.
Thanks for the input, and please just let me know if I can support the work in some way!